src/opm/output/eclipse/AggregateNetworkData.cpp

```diff
@@ -0,0 +1,752 @@
+#include <opm/output/eclipse/WriteRestartHelpers.hpp>
+#include <opm/output/eclipse/VectorItems/group.hpp>
+#include <opm/output/eclipse/VectorItems/network.hpp>
+#include <opm/output/eclipse/VectorItems/intehead.hpp>
+
+#include <opm/parser/eclipse/EclipseState/EclipseState.hpp>
+#include <opm/parser/eclipse/EclipseState/Runspec.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/SummaryState.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/Schedule.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/Group/GTNode.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/Group/Group.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/Network/ExtNetwork.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/Well/Well.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/Network/Branch.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/Network/Node.hpp>
+
+#include <opm/parser/eclipse/Units/UnitSystem.hpp>
+#include <opm/parser/eclipse/Units/Units.hpp>
+
+
+#include <algorithm>
+#include <cstddef>
+#include <cstring>
+#include <exception>
+#include <string>
+#include <stdexcept>
+
+#define ENABLE_GCNTL_DEBUG_OUTPUT 0
+
+#if ENABLE_GCNTL_DEBUG_OUTPUT
+#include <iostream>
+#endif // ENABLE_GCNTL_DEBUG_OUTPUT
+
+// #####################################################################
+// Class Opm::RestartIO::Helpers::AggregateNetworkData
+// ---------------------------------------------------------------------
+
+namespace VI = Opm::RestartIO::Helpers::VectorItems;
+
+namespace {
+
+// maximum number of network nodes
+std::size_t nodmax(const std::vector<int>& inteHead)
+{
+    return inteHead[Opm::RestartIO::Helpers::VectorItems::NODMAX];
+}
+
+// maximum number of network branches
+std::size_t nbrmax(const std::vector<int>& inteHead)
+{
+    return inteHead[Opm::RestartIO::Helpers::VectorItems::NBRMAX];
+}
+
+std::size_t entriesPerInobr(const std::vector<int>& inteHead)
+{
+    return inteHead[Opm::RestartIO::Helpers::VectorItems::NINOBR];
+}
+
+template <typename NodeOp>
+void nodeLoop(const std::vector<const std::string*>& nodeNamePtrs,
+              NodeOp&&                             nodeOp)
+{
+    auto nodeID = std::size_t {0};
+    for (const auto* nodeNmPtr : nodeNamePtrs) {
+        nodeID += 1;
+
+        if (nodeNmPtr == nullptr) {
+            continue;
+        }
+
+        nodeOp(*nodeNmPtr, nodeID - 1);
+    }
+}
+
+template <typename BranchOp>
+void branchLoop(const std::vector<const Opm::Network::Branch*>& branches,
+                BranchOp&&                             branchOp)
+{
+    auto branchID = std::size_t {0};
+    for (const auto* branch : branches) {
+        branchID += 1;
+
+        if (branch   == nullptr) {
+            continue;
+        }
+
+        branchOp(*branch, branchID - 1);
+    }
+}
+
+template < typename T>
+std::pair<bool, int > findInVector(const std::vector<T>  & vecOfElements, const T  & element)
+{
+    std::pair<bool, int > result;
```

Comment:
No - this is what we have `std::optional<int>` for.

Reply:
Fine -OK, changed.